### PR TITLE
Capybara test

### PR DESCRIPTION
**Pull request by rbd3**
# Change made:
👉 Write integration tests with Capybara gem.
👉 Secure app from n+1 problems.

N+1-before

N+1-after

### CODE REVIEW

File: spec/views/users/show.html.erb_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  scenario 'Redirecting to Post Show Page' do
+    visit user_path(user)
+
+    # Go back to the user's show page
+    visit user_path(user)
+
+    click_link 'See All Posts'
+    expect(current_path).to eq(user_posts_path(user))
+  end
```
Comment (TedLivist) [inline spec/views/users/show.html.erb_spec.rb:35]: - [ ] Kindly add test to assert that when a post is clicked it redirects to the post show page.

File: spec/views/posts/index.html.erb_spec.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    expect(page).to have_content('this is body of a post')
+  end
+
+  scenario 'Displays title of the post correctly' do
+    visit user_posts_path(user)
+
+    expect(page).to have_content('Post 1')
+    expect(page).to have_content('Post 2')
+  end
```
Comment (TedLivist) [inline spec/views/posts/index.html.erb_spec.rb:55]: - [ ] Kindly write tests to assert we can see a section for pagination if there are more posts than fit on the view.

- [ ] Kindly write test to that redirects to the post's show page when you click on a post.

File: spec/views/posts/show.html.erb_spec.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    visit user_post_path(user_id: post1.author_id, id: post1.id)
+
+    expect(page).to have_content('John')
+  end
+
+  scenario 'Displays comment left by the commenter' do
+    visit user_post_path(user_id: post1.author_id, id: post1.id)
+
+    expect(page).to have_content('Comment 1 by John')
+  end
```
Comment (TedLivist) [inline spec/views/posts/show.html.erb_spec.rb:37]: - [ ] Kindly add tests to assert you can see the post's title.

- [ ] Kindly add tests to assert you can see the post body.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @rbd3 and @noelfoka,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### Kindly note that the owner of the repo should be the one who submits the form for alignment of reviews.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by julie-ify — COMMENTED** [review]
Hi @rbd3 and @noelfoka,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

* For this project, you are required to build the project afresh for each partner's repo, so you cannot request a first review on this PR anymore. Kindly work on your other partner's repo and submit as a new review, or submit the correct CR form.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by mckent05 — COMMENTED** [review]
Hi team, 

# STATUS: INVALID REQUEST ❌ ❌ 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project that does not belong to you. Please note that though this is a pair programming project, you are required to pair-program on your individual repos and submit for a review individually. @noelfoka, please note that you are required to submit your own repo for review. Please note this 👍 👍 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by Nemwel-Boniface — COMMENTED** [review]
## @rbd3  and @noelfoka :wave: 

I would like you both to know that we are aware of the error that was made by the first reviewer @TedLivist . This is a pair programming task which should be submitted solo which means that @noelfoka  you should work on the same project on your repo together with @rbd3  and request for a review using your own repo.

**What needs to be done:**
1. @rbd3  please feel free to request for a review and you will receive a review from the reviewers.
2. @noelfoka  kindly work on the project on your own end and when you are ready, you can feel free to request for a review from your dashboard.
3. Also @noelfoka  the first review you received from @TedLivist  will be invalidated soon and you will be able to submit your repo when you are ready as a first review and not a re-review.


Incase anything is not clear please tag me @Nemwel-Boniface so that I get the notification. :rocket:

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @rbd3, and Team👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Good job fixing N+1 problems. ✔️
- Good job with integration specs. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️

- [ ] The issues page link is broken in the README file. Please make sure to fix this. It will make the README file more professional.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by V-Blaze — APPROVED** [review]
Hi @rbd3,

# Approved 🟢 🟢 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy

## Highlights

- Fixed N+1 problems.💯
-  Readme looks professional 👍
-  No broken links 🟢 👍

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**